kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaProtocolHandler.java

```diff
@@ -550,10 +336,34 @@ private GroupCoordinator createAndBootGroupCoordinator(String tenant) {
             groupCoordinator = startGroupCoordinator(tenant, offsetTopicClient);
 
             // and listener for Offset topics load/unload
-            brokerService.pulsar()
-                    .getNamespaceService()
-                    .addNamespaceBundleOwnershipListener(
-                            new OffsetAndTopicListener(brokerService, tenant, kafkaConfig, groupCoordinator));
+            final NamespaceName kafkaMetaNs = NamespaceName.get(tenant, kafkaConfig.getKafkaMetadataNamespace());
+            final NamespaceName kafkaTopicNs = NamespaceName.get(tenant, kafkaConfig.getKafkaNamespace());
+            final String metadataNamespace = kafkaConfig.getKafkaMetadataNamespace();
+            bundleListener.addTopicOwnershipListener(new TopicOwnershipListener() {
+                @Override
+                public void whenLoad(TopicName topicName) {
+                    if (KopTopic.isGroupMetadataTopicName(topicName.toString(), metadataNamespace)) {
+                        groupCoordinator.handleGroupImmigration(topicName.getPartitionIndex());
+                    }
+                }
+
+                @Override
+                public void whenUnload(TopicName topicName) {
+                    if (KopTopic.isGroupMetadataTopicName(topicName.toString(), metadataNamespace)) {
+                        groupCoordinator.handleGroupEmigration(topicName.getPartitionIndex());
+                    }
+                }
+
+                @Override
+                public String name() {
+                    return "OffsetAndTopicListener";
+                }
+
+                @Override
+                public boolean test(NamespaceName namespaceName) {
+                    return namespaceName.equals(kafkaMetaNs) || namespaceName.equals(kafkaTopicNs);
+                }
```

Comment:
the same as above, for `group metadata topics`, we do not need to judge whether `kafkaTopicNs` is hit.